src/containers/repository.py

```diff
@@ -21,3 +46,201 @@ def load_from_json(self, repository):
+
+                    # Logging the number of loaded issues
+                    # TODO: Logic for printing will be removed, due to logging refactoring later
+                    if label_name is None:
+                        print(f"Loaded {len(repository_issues)} issues.")
+                    else:
+                        print(f"Loaded {len(repository_issues)} issues for label `{label_name}`.")
+
+                    # If we retrieved less than issue_per_page constant, it means we're on the last page
+                    if len(issues_json) < ISSUE_PER_PAGE:
+                        break
+                    page += 1
+
+            # Specific error handling for HTTP errors
+            except requests.HTTPError as http_err:
+                print(f"HTTP error occurred: {http_err}")
+
+            except Exception as e:
+                print(f"An error occurred: {e}")
+                break
+
+        return loaded_issues
+
+    def get_base_endpoint(self, label_name: str) -> str:
+        """
+        Prepares the base endpoint for fetching issues from a GitHub repository.
+        If the label name is not specified, the base endpoint is for fetching all issues.
+
+        @param label_name: The name of the label.
+
+        @return: The base endpoint for fetching issues.
+        """
+        if label_name is None:
+            search_query = f"repo:{self.organization_name}/{self.repository_name} is:issue"
+        else:
+            search_query = f"repo:{self.organization_name}/{self.repository_name} is:issue label:{label_name}"
+
+        base_endpoint = f"https://api.github.com/search/issues?q={search_query}&per_page={ISSUE_PER_PAGE}"
+
+        return base_endpoint
+
+    def get_unique_projects(self, session: requests.sessions.Session, projects_title_filter: str, projects: dict):
+        """
+        Generate a main structure for every unique project.
+        Connects project with the repositories.
+
+        @param projects: The dictionary to store project objects.
+        @param projects_title_filter: The list of project titles to filter.
+        @param session: A configured request session.
+
+        @return: The unique project structure as a dictionary.
+        """
+
+        # Fetch all projects, that are attached to the repo
+        gh_projects = self.get_gh_projects(session)
+
+        # Update unique_projects with main project structure for every project
+        for gh_project in gh_projects:
+            subscriptable_project = gh_project.to_dict()
+            project_id = subscriptable_project["id"]
+            project_title = subscriptable_project["title"]
+
+            is_project_required = (projects_title_filter == '[]') or (project_title in projects_title_filter)
+
+            if is_project_required:
+                if project_id not in projects:
+                    project = Project()
+                    project.load_from_api_json(self, subscriptable_project)
```

Comment:
Here I would still send gh_project. The conversion to dict can be done when really needed.

Reply:
Solved during refactoring phases.